File: Data Structures & Algorithms/sort-colors/submission-3.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int r = 0;
        int l = nums.size() - 1;

        int i = 0;
        while (i <= l) {
            if (nums[i] == 0) {
                swap(nums[i], nums[r]);
                r++;
            }
            
            if (nums[i] == 2) {
                swap(nums[i], nums[l]);
                l--;
                continue;
            } 
            
            i++;
        }
    }
};

File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)-1
        while l <= r:
            m = l + ((r-l)//2)
            if nums[m] == target:
                return m
            elif m == l:
                break
            elif nums[m] < target:
                l = m
            elif nums[m] > target:
                r = m
        return -1 if nums[r] != target else r